docs/quic-channels.md

```diff
@@ -0,0 +1,35 @@
+# Aranya Quic Channels
+
+## Overview
+
+Aranya Quic Channels are a way for application data to be sent over the 
+network to other peers using TLS to ensure security. The QUIC protocol will
+be used to send data. This provides multiplexing of connections, tls for
+security, and custom congestion control.
+
+## Channels
+
+Quic channels will be stored in an `FnvIndexMap` with a max size of `32`. 
+Creating a channel when the map is full will fail with a `QuicChannelError`.
+
+```rust
+/// FNVIndexMap requires that the size be a power of 2.
+const MAXIMUM_CHANNELS: usize = 32;
+type ChannelMap = FNVIndexMap<(SocketAddr, GraphID), BiDirectionalStream, MAXIMUM_CHANNELS>
+```
+
+BiDirectional streams will have the following methods, mirroring the API of
+s2n_quic. Receive will return a chunk of data from the stream. It will 
+return Ok(None) when the stream is finished.
+
+```rust
+pub async fn receive(&mut self) -> Result<Option<bytes::Bytes>>
```

Comment:
`receive` should write _into_ a buffer, like `recv(&mut self, buf: &mut [u8]) -> Result<usize, Error>`